ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java

```diff
@@ -606,13 +607,37 @@ public void processDeposits(MutableBeaconState state, SszList<? extends Deposit>
       throws BlockProcessingException {
     safelyProcess(
         () -> {
+          final boolean depositSignaturesAreAllGood = batchVerifyDepositSignatures(deposits);
```

Comment:
Things like the fuzzer disable deposit signature verification entirely to speed up processing. So we probably want to move that check to here since the net result is that we treat all signatures as valid.

```suggestion
          final boolean depositSignaturesAreAllGood = !blsVerifyDeposit || batchVerifyDepositSignatures(deposits);
```

And then we can remote the `blsVerifyDeposit` check from `depositSignatureIsValid`.

Reply:
Good catch!

It seems difficult to remove the check from `depositSignatureIsValid` since the `BlockValidatorTest` class sets `blsVerifyDeposit` to false and ends up calling `processDepositWithoutCheckingMerkleProof` directly via `GenesisGenerator`. That is, it bypasses the batch verification.